src/sumologic-span-processor/document-visibility-state.ts

```diff
@@ -55,7 +60,14 @@ window.addEventListener('pageshow', () => {
 });
 
 export const onStart = (span: SdkTraceSpan, context?: Context): void => {
-  span.setAttribute(ATTRIBUTE_NAME, currentState);
+  const startTimeInNanoseconds = hrTimeToNanoseconds(span.startTime);
+  for (let i = changes.length - 1; i >= 0; i -= 1) {
+    const { timestampInNanoseconds, state } = changes[i];
+    if (timestampInNanoseconds <= startTimeInNanoseconds) {
+      span.setAttribute(ATTRIBUTE_NAME, state);
+      break;
+    }
+  }
```

Comment:
Was the issue that the visibility state was taken from an incorrect, too late moment?

Reply:
The main issue is that `documentLoad` span is created as soon as browser tab opens. Our RUM script is evaluated after some time. In this PR we make sure, that `documentLoad` span is populated with proper `document.visibilityState` attribute. Previously we considered the `onStart` function invocation time, not the given span start time.

Reply:
Some comments would be useful here. :)

Reply:
Added explanation.